Interface.C

```diff
@@ -430,6 +439,15 @@ void preciceAdapter::Interface::writeCouplingData()
                 numDataLocations_,
                 vertexIDs_,
                 dataBuffer_);
+
+            if (precice_.isGradientDataRequired(couplingDataWriter->dataID()))
+            {
+                couplingDataWriter->writeGradients(gradientBuffer_, dim_);
+                precice_.writeBlockVectorGradientData(couplingDataWriter->dataID(),
+                                                      numDataLocations_,
+                                                      vertexIDs_,
+                                                      gradientBuffer_.data());
```

Comment:
One example for vector data would be velocity. We should also implement that (in this or in a follow-up PR), otherwise this case will be completely unused for now.